app/src/test/java/io/intrepid/contest/screens/contestcreation/NewContestPresenterTest.java

```diff
@@ -0,0 +1,102 @@
+package io.intrepid.contest.screens.contestcreation;
+
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.intrepid.contest.models.Category;
+import io.intrepid.contest.models.Contest;
+import io.intrepid.contest.testutils.TestPresenterConfiguration;
+
+import static junit.framework.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NewContestPresenterTest {
+
+    @Mock
+    NewContestMvpContract.View mockView;
+    @Mock
+    ContestCreationFragment mockChildfFragment;
+    private List<Category> categories;
+    private NewContestPresenter newContestPresenter;
+
+    @Before
+    public void setup(){
+        MockitoAnnotations.initMocks(this);
+        newContestPresenter = new NewContestPresenter(mockView, TestPresenterConfiguration.createTestConfiguration());
+        when(mockView.getChildEditFragment(anyInt())).thenReturn(mockChildfFragment);
+        newContestPresenter.onViewCreated();
+
+        categories = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
+            categories.add(new Category("TEST TITLE " + i, "TEST DESCRIPTION " + i));
+        }
+    }
+
+    @Test
+    public void presenterCanBeCreatedSuccessfully() {
+        assertTrue(newContestPresenter != null);
+        assertTrue(newContestPresenter.contest != null);
+        verify(mockView).showContestSubmissionPage(0);
+    }
+
+    @Test
+    public void firstBackNavigationShouldCancelEdit() {
+        newContestPresenter.onBackButtonClicked();
+        verify(mockView).cancelEdit();
+    }
+
+    @Test
+    public void legalBackNavigationShouldMoveBackward() {
+        newContestPresenter.setContestDescription(" d ");
+        newContestPresenter.onBackButtonClicked();
+        verify(mockView).showContestSubmissionPage(1);
+    }
+
+    @Test
+    public void forwardNavigationShouldMoveForward() {
+        newContestPresenter.onNextButtonClicked();
+        verify(mockChildfFragment).onNext();
+    }
+
+    @Test
+    public void setContestNameShouldModifyContestName() {
```

Comment:
What about some spaces between the 3 steps?
(newline after 75, new line after 76)